Mathlib/Analysis/Normed/Algebra/Exponential.lean

```diff
@@ -413,136 +435,157 @@ variable [CompleteSpace 𝔸]
   classical
     induction' s using Finset.induction_on with a s ha ih
     · simp
-    rw [Finset.noncommProd_insert_of_not_mem _ _ _ _ ha, Finset.sum_insert ha, exp_add_of_commute,
+    rw [Finset.noncommProd_insert_of_not_mem _ _ _ _ ha, Finset.sum_insert ha, exp_add_of_commute 𝕂,
       ih (h.mono <| Finset.subset_insert _ _)]
     refine Commute.sum_right _ _ _ fun i hi => ?_
     exact h.of_refl (Finset.mem_insert_self _ _) (Finset.mem_insert_of_mem hi)
 
-theorem exp_nsmul (n : ℕ) (x : 𝔸) : exp 𝕂 (n • x) = exp 𝕂 x ^ n := by
+include 𝕂 in
+theorem exp_nsmul (n : ℕ) (x : 𝔸) : exp (n • x) = exp x ^ n := by
   induction n with
   | zero => rw [zero_smul, pow_zero, exp_zero]
-  | succ n ih => rw [succ_nsmul, pow_succ, exp_add_of_commute ((Commute.refl x).smul_left n), ih]
-
-variable (𝕂)
+  | succ n ih => rw [succ_nsmul, pow_succ, exp_add_of_commute 𝕂 ((Commute.refl x).smul_left n), ih]
 
+include 𝕂 in
 /-- Any continuous ring homomorphism commutes with `NormedSpace.exp`. -/
 theorem map_exp {F} [FunLike F 𝔸 𝔹] [RingHomClass F 𝔸 𝔹] (f : F) (hf : Continuous f) (x : 𝔸) :
-    f (exp 𝕂 x) = exp 𝕂 (f x) :=
-  map_exp_of_mem_ball f hf x <| (expSeries_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _
+    f (exp x) = exp (f x) :=
+  map_exp_of_mem_ball 𝕂 f hf x <| (expSeries_radius_eq_top 𝕂 𝔸).symm ▸ edist_lt_top _ _
 
+include 𝕂 in
 theorem exp_smul {G} [Monoid G] [MulSemiringAction G 𝔸] [ContinuousConstSMul G 𝔸] (g : G) (x : 𝔸) :
-    exp 𝕂 (g • x) = g • exp 𝕂 x :=
+    exp (g • x) = g • exp x :=
   (map_exp 𝕂 (MulSemiringAction.toRingHom G 𝔸 g) (continuous_const_smul g) x).symm
 
-theorem exp_units_conj (y : 𝔸ˣ) (x : 𝔸) : exp 𝕂 (y * x * ↑y⁻¹ : 𝔸) = y * exp 𝕂 x * ↑y⁻¹ :=
-  exp_smul _ (ConjAct.toConjAct y) x
-
-theorem exp_units_conj' (y : 𝔸ˣ) (x : 𝔸) : exp 𝕂 (↑y⁻¹ * x * y) = ↑y⁻¹ * exp 𝕂 x * y :=
-  exp_units_conj _ _ _
-
-@[simp]
-theorem _root_.Prod.fst_exp [CompleteSpace 𝔹] (x : 𝔸 × 𝔹) : (exp 𝕂 x).fst = exp 𝕂 x.fst :=
-  map_exp _ (RingHom.fst 𝔸 𝔹) continuous_fst x
-
-@[simp]
-theorem _root_.Prod.snd_exp [CompleteSpace 𝔹] (x : 𝔸 × 𝔹) : (exp 𝕂 x).snd = exp 𝕂 x.snd :=
-  map_exp _ (RingHom.snd 𝔸 𝔹) continuous_snd x
-
-@[simp]
-theorem _root_.Pi.coe_exp {ι : Type*} {𝔸 : ι → Type*} [Finite ι] [∀ i, NormedRing (𝔸 i)]
-    [∀ i, NormedAlgebra 𝕂 (𝔸 i)] [∀ i, CompleteSpace (𝔸 i)] (x : ∀ i, 𝔸 i) (i : ι) :
-    exp 𝕂 x i = exp 𝕂 (x i) :=
-  let ⟨_⟩ := nonempty_fintype ι
-  map_exp _ (Pi.evalRingHom 𝔸 i) (continuous_apply _) x
-
-theorem _root_.Pi.exp_def {ι : Type*} {𝔸 : ι → Type*} [Finite ι] [∀ i, NormedRing (𝔸 i)]
-    [∀ i, NormedAlgebra 𝕂 (𝔸 i)] [∀ i, CompleteSpace (𝔸 i)] (x : ∀ i, 𝔸 i) :
-    exp 𝕂 x = fun i => exp 𝕂 (x i) :=
+include 𝕂 in
+theorem exp_units_conj (y : 𝔸ˣ) (x : 𝔸) : exp (y * x * ↑y⁻¹ : 𝔸) = y * exp x * ↑y⁻¹ :=
+  exp_smul 𝕂 (ConjAct.toConjAct y) x
+
+include 𝕂 in
+theorem exp_units_conj' (y : 𝔸ˣ) (x : 𝔸) : exp (↑y⁻¹ * x * y) = ↑y⁻¹ * exp x * y :=
+  exp_units_conj 𝕂 _ _
+
+-- @[simp]
```

Comment:
Why? Can't we have this?

Reply:
Because the proof needs you to pick a `𝕂`, and `simp` doesn't know how to do that now that it can't choose it by unification.